users/views.py

```diff
@@ -46,21 +46,151 @@ def post(self, request):
+            user_data = User.objects.get(roll = roll_no)
+            sender = EMAIL_HOST_USER
+            recipient = user_data.email
+            name = user_data.name
+            user_code = user_data.generate_verification_code()
+            user_link = EMAIL_LINK["ForgotPass"].format(code = user_code)
+            subject = EMAIL_SUBJECT["ForgotPass"]
+            body = EMAIL_BODY["ForgotPass"].format(name=name, link=user_link)
+            send_mail(subject, body, sender, [recipient], fail_silently=False)
+            return redirect(REDIRECT_LINK["ForgotPass"])
+        except:
+            return HttpResponse("Please set up email host details!", status=206)
+    else :
+        return HttpResponse("Invalid request!", status=400)
+@api_view(['POST'])
+def ForgotPass(request,token):
+    if request.method == "POST":
+        try:
+            user_data=User.objects.get(verification_code=token)
+            if user_data.activated==True:
+                user_data.activated=False
+                user_data.save()
+                response={
+                    'status':'success',
+                    'code':status.HTTP_200_OK,
+                    'message':'Account successfully deactivated. Now follow activation process to create new password and activate account',                }
+                return Response(response)
+            else:
+                response={
+                'code':'status.HTTP_401_UNAUTHORIZED',
+                'message':'Token already used'}
+                return Response(response,status=401)  
+        except:
+            response={
+                    'code':'status.HTTP_401_UNAUTHORIZED',
+                    'message':'Invalid token or invalid request'
+                }
+            return Response(response,status=status.HTTP_401_UNAUTHORIZED)
+    else:
+        return HttpResponse("Invalid Request",status=400)
+
+#
+
+#Create update fblink API
+class FbLinkView(APIView):
+
+    def post(self, request):
+        user = IsLoggedIn(request)
+        if user is not None:
+            try:
+                fb_link = request.data.get("fblink", "")
+                user.fblink=fb_link
+                user.save()
+                return Response(status = status.HTTP_200_OK)
+            
+            except:
+                return Response(status = status.HTTP_400_BAD_REQUEST)
+        
+        else:
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
+#
+
+#Create Friend's profile(by name) API
+class PeopleProfileViewName(APIView):
+
+    def post(self,request):
+        try:
+            username = request.data.get("username", "")
+            user = User.objects.get(username=username)
+            if user is None:
+                return HttpResponse("User does not exist.",status=401)
+            serializer = UserViewSerializer(user)
+            return Response(serializer.data,status=status.HTTP_200_OK)
+        except:
+            return HttpResponse("Bad Request.",status=400)
```

Comment:
Please change the HttpResponses to Response with the appropriate messages.

Reply:
ok